text/0000-float-semantics.md

```diff
@@ -0,0 +1,255 @@
+- Feature Name: `float_semantics`
+- Start Date: 2023-10-14
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Rust's floating point operations follow IEEE 754-2008 -- with some caveats around operations producing NaNs: IEEE makes almost no guarantees about the sign and payload bits of the NaN; however, actual hardware does not pick those bits completely arbitrarily, and Rust will expose some of those hardware-provided guarantees to programmers.
+On the flip side, NaN generation is non-deterministic: running the same operation on the same inputs several times can produce different results.
+And there is a caveat: while IEEE specifies that float operations can never output a signaling NaN, Rust float operations *can* produce signaling NaNs, *but only if* an input is signaling.
+That means the only way to ever see a signaling NaN in a program is to create one with `from_bits` (or equivalent unsafe operations).
```

Comment:
From a reader's perspective, this part doesn't differentiate 'current behavior' and 'proposed behavior' well?

Reply:
The RFC describes the proposed spec.

Current situation is that we have to spec so there's not really anything to compare with.
Current behavior as implemented by rustc already satisfies the proposed spec.